dll/cpl/appwiz/createlink.c

```diff
@@ -299,16 +324,56 @@ FinishDlgProc(HWND hwndDlg,
             pContext = (PCREATE_LINK_CONTEXT) GetWindowLongPtr(hwndDlg, DWLP_USER);
             if (lppsn->hdr.code == PSN_WIZFINISH)
             {
-                SendDlgItemMessageW(hwndDlg, IDC_SHORTCUT_NAME, WM_GETTEXT, MAX_PATH, (LPARAM)pContext->szDescription);
-                wcscat(pContext->szLinkName, pContext->szDescription);
-                wcscat(pContext->szLinkName, L".lnk");
-                if (!CreateShortcut(pContext))
+                LPWSTR pch;
+                DWORD attrs;
+                GetDlgItemTextW(hwndDlg, IDC_SHORTCUT_NAME, pContext->szDescription, MAX_PATH);
+                StrTrimW(pContext->szDescription, L" \t");
+
+                /* if old shortcut file exists, then delete it now */
+                attrs = GetFileAttributesW(pContext->szOldFile);
+                if (attrs != INVALID_FILE_ATTRIBUTES && !(attrs & FILE_ATTRIBUTE_DIRECTORY))
                 {
-                    MessageBox(hwndDlg, _T("Failed to create shortcut"), _T("Error"), MB_ICONERROR);
+                    DeleteFileW(pContext->szOldFile);
                 }
-            }
-
 
+                if (IsInternetLocation(pContext->szTarget))
+                {
+                    /* internet */
+                    StringCchCopyW(pContext->szLinkName, _countof(pContext->szLinkName),
+                                   pContext->szOrigin);
+                    PathAppendW(pContext->szLinkName, pContext->szDescription);
+
+                    /* change extension if any */
+                    pch = PathFindExtensionW(pContext->szLinkName);
+                    if (pch && *pch)
+                        *pch = 0;
+                    StringCchCatW(pContext->szLinkName, _countof(pContext->szLinkName), L".url");
+
+                    if (!CreateInternetShortcut(pContext))
+                    {
+                        MessageBox(hwndDlg, _T("Failed to create internet shortcut"), NULL, MB_ICONERROR);
```

Comment:
MessageBoxW, and use L"..." for the string.

Reply:
OK.